packages/vite/src/node/plugins/asset.ts

```diff
@@ -431,8 +431,12 @@ function svgToDataURL(content: Buffer): string {
       stringContent
         .trim()
         .replaceAll(/>\s+</g, '><')
-        .replaceAll('"', "'")
         .replaceAll('%', '%25')
+        .replaceAll(
+          /"([^"']*'[^"]*)"/g,
+          (_, capture) => `%22${capture.replaceAll("'", '%27')}%22`,
+        )
+        .replaceAll('"', "'")
```

Comment:
I think this cannot handle this case: `data-name1='so"me' data-name2='so"me'`
I guess we need a lexer to handle that.
I think it's easier to fallback to base64 if the SVG includes both `"` and `'`.

Reply:
I totally agree with you. And sorry I missed that case. I can detect that condition you mentioned. However, for the lexer, is there a util function that I can use or should I use builtin `encodeURI()`?

Reply:
I thought of the base64 as the one below.

https://github.com/vitejs/vite/blob/bb07f605cca698a81f1b4606ddefb34485069dd1/packages/vite/src/node/plugins/asset.ts#L427